src/vs/editor/contrib/suggest/suggestModel.ts

```diff
@@ -95,6 +97,20 @@ export const enum State {
 	Auto = 2
 }
 
+function shouldPreventQuickSuggest(contextKeyService: IContextKeyService, configurationService: IConfigurationService): boolean {
+	return (
+		Boolean(contextKeyService.getContextKeyValue<boolean>('inlineSuggestionVisible'))
+		&& !Boolean(configurationService.getValue<boolean>('editor.inlineSuggest.allowQuickSuggestions'))
```

Comment:
For a small change you could be reading the config via the context key service, e.g `config.editor.inlineSuggest.allowQuickSuggestions`

Reply:
I think it's fine, it was more for fyi